CIScripts/Test/Tests/Protocol/TunnellingWithAgent.Tests.ps1

```diff
@@ -371,37 +389,48 @@ Describe "Tunnelling with Agent tests" {
     Context "IP fragmentation" {
         # TODO: Enable this test once fragmentation is properly implemented in vRouter
         It "ICMP - Ping with big buffer succeeds" -Pending {
-            Test-Ping `
-                -Session $Sessions[0] `
-                -SrcContainerName $Container1ID `
-                -DstContainerName $Container2ID `
-                -DstContainerIP $Container2NetInfo.IPAddress `
-                -BufferSize 1473 | Should Be 0
-
-            Test-Ping `
-                -Session $Sessions[1] `
-                -SrcContainerName $Container2ID `
-                -DstContainerName $Container1ID `
-                -DstContainerIP $Container1NetInfo.IPAddress `
-                -BufferSize 1473 | Should Be 0
+            $Container1MaxBufferSize = Get-MaxICMPDataSizeForMTU -MTU $Container1NetInfo.MtuSize
+            $Container2MaxBufferSize = Get-MaxICMPDataSizeForMTU -MTU $Container2NetInfo.MtuSize
+
+            $SrcContainers = @($Container1ID, $Container2ID)
+            $DstContainers = @($Container2ID, $Container1ID)
+            $DstIPs = @($Container2NetInfo.IPAddress, $Container1NetInfo.IPAddress)
+            $BufferSizes = @($Container1MaxBufferSize, $Container2MaxBufferSize)
+
+            foreach ($ContainerIdx in @(0, 1)) {
+                $BufferSizeLargerBefore = $BufferSizes[$ContainerIdx] + 1
+                $BufferSizeLargerAfter = $BufferSizes[$ContainerIdx] - 1
+                foreach ($BufferSize in @($BufferSizeLargerBefore, $BufferSizeLargerAfter)) {
+                    Test-Ping `
+                        -Session $Sessions[$ContainerIdx] `
+                        -SrcContainerName $SrcContainers[$ContainerIdx] `
+                        -DstContainerName $DstContainers[$ContainerIdx] `
+                        -DstContainerIP $DstIPs[$ContainerIdx] `
+                        -BufferSize $BufferSize | Should Be 0
+                }
+            }
         }
 
         # TODO: Enable this test once fragmentation is properly implemented in vRouter
         It "UDP - sending big buffer succeeds" -Pending {
-            $MyMessage = "buffer" * 300
+            $MaxBufferSize = Get-MaxUDPDataSizeForMTU -MTU $Container1NetInfo.MtuSize
             $UDPServerPort = 1111
             $UDPClientPort = 2222
 
-            Test-UDP `
-                -Session1 $Sessions[0] `
-                -Session2 $Sessions[1] `
-                -Container1Name $Container1ID `
-                -Container2Name $Container2ID `
-                -Container1IP $Container1NetInfo.IPAddress `
-                -Container2IP $Container2NetInfo.IPAddress `
-                -Message $MyMessage `
-                -UDPServerPort $UDPServerPort `
-                -UDPClientPort $UDPClientPort | Should Be $true
+            $MessageLargerBefore = "buffer" * $MaxBufferSize
```

Comment:
What happens if we exceed `$MaxBufferSize`? We write to unallocated memory? It's not clear from the name.

Maybe a better name would be `$MessageFragmentationThreshold` or `$MaxMsgSizeWithoutFragmentation` or something?